features/step_definitions/test_steps.rb

```diff
@@ -7,7 +7,7 @@
   add_named_record test_key, key
 
   date = if days
-    days.split(' ')[0].to_i.days.ago
+    days.split(' ')[0].to_i.send(interval).ago
```

Comment:
The `split` is superfluous given the change made to the regexp. Also, I would rename the variable `interval_count` or something other than `days`.